### Lab: Hypothesis Testing
It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

- Set up the hypothesis test.
- Write down all the steps followed for setting up the test.
- Calculate the test statistic by hand and also code it in Python. It should be 4.76190. What decision can you make based on this calculated value?


In [11]:
import numpy as np
import pandas as pd
import scipy.stats as stats
from scipy.stats import t

#### 6 Analysis Steps
Step 1: Determine H0 and H1 (Ha)<br>
Step 2: Investigate sample size n >= 30 or n < 30. Sample size is n >= 30, but as σ of population is unknown we use T test statistic.<br>
Step 3: Calculate test statistic --> we decided for T: tc = (x̄ - μ) / (S /$\sqrt{n}$)<br>
Step 4: Calculate tc critical value to derive the interval<br>
Step 5: Decide if H0 can be rejected or we fail to reject H0<br>
Step 6: Determine P value math. 

#### 1. Setup of hypothesis test
**Step 1:** <br>
H0 = The mean blood pressure in Honolulu Heart Study does not show a significant difference to the mean systolic blood pressure: **μ != 120 mm Hg**. </br>
H1 = The mean blood pressure in Honolulu Heart Study (**x̄ = 130.1 mm Hg**) shows a significant difference to the mean systolic blood pressure which is **μ = 120 mm Hg**.<br>
We decided for a 2-sided test.

#### Step 2:
Sample size:     n = 100<br>
Population mean: μ = 120 mm Hg<br>
Sample mean:     x̄ = 130.1 mm Hg<br>
Sample std. dev.: S = 21.21 mm Hg<br>
Std. dev. of population is unknown: σ = ?<br>

As σ of population is not known we choose t-test statistic for step 3.

#### Step 3: Calculation of the test statistic of the sample

In [18]:
# Calculation of the t-test statistic of the sample
# (x̄ - μ) / (S / 100^1/2) 
t_sample = (130.1-120)/(21.21/(100**(1/2)))
print('t_sample = +/- {:.4f}'.format(t_sample))

t_sample = +/- 4.7619


#### Step 4:
#### Calculation of critical interval

In [15]:
# 2-sided student test
tc = scipy.stats.t.ppf(1-(0.05/2), df=99) #can be also written like: tc = scipy.stats.t.ppf(0.975, df=99)
print('tc = +/-{:.4f}'.format(tc))

# 2-sided normal distribution test with confidence level 95% which is the same as saying 𝛼 = 1−0.95=0.05
zc = stats.norm.ppf(0.975)
print('zc = +/-{:.4f}'.format(zc))

tc = +/-1.9842
zc = +/-1.9600


#### Step 5:
H0 has to be rejected: the t_sample = +/- 4.7619 is not in the critical interval.
H1 has to be accepted: the t_sample is significantly different to the mean of the systolic blood pressure.

#### Step 6

In [36]:
tc = scipy.stats.t.ppf(0.99999671, df=99)
P = (1-0.99999671)*100
print('P of t_sample = +/- {:.5f}%'.format(P))
tc

P of t_sample = +/- 0.00033%


4.761251569030441